File: Innovation-Challenge/张秋伟-RickAndMorty角色扮演/kangaroo_flutter/base_lib/lib/tools/store/filestore/storage_type.dart


enum StorageType {
  typeLog,
  typeCache,
  typeTemp,
  typeFile,
  typeAudio,
  typeImage,
  typeVideo,
  typeThumbImage,
  typeThumbVideo,
}

extension StorageTypeExtension on StorageType{
  String get storageDirectoryName{
    switch(this){
      case StorageType.typeLog:
        return Storage.log;
      case StorageType.typeCache:
        return Storage.cache;
      case StorageType.typeTemp:
        return Storage.temp;
      case StorageType.typeFile:
        return Storage.file;
      case StorageType.typeAudio:
        return Storage.audio;
      case StorageType.typeImage:
        return Storage.image;
      case StorageType.typeVideo:
        return Storage.video;
      case StorageType.typeThumbImage:
        return Storage.thumb;
      case StorageType.typeThumbVideo:
        return Storage.thumb;
    }
  }
}

class Storage{
  static const String audio = "audio/";
  static const String data = "data/";
  static const String file = "file/";
  static const String log = "log/";
  static const String cache = "cache/";
  static const String temp = "temp/";
  static const String image = "image/";
  static const String thumb = "thumb/";
  static const String video = "video/";
}

File: Innovation-Challenge/史大伟-AI招聘助手/kangaroo_lib_for_kotlin/ktlib/src/main/java/com/kangaroo/ktlib/util/store/filestorage/StorageType.kt
package com.kangaroo.ktlib.util.store.filestorage

import com.kangaroo.ktlib.util.store.filestorage.UStorage.THRESHOLD_MIN_SPCAE

enum class StorageType @JvmOverloads constructor ( val storageDirectoryName : DierctoryType,val storageMinSize:Long = THRESHOLD_MIN_SPCAE) {
    TYPE_LOG(DierctoryType.log),
    TYPE_CACHE(DierctoryType.cache),
    TYPE_TEMP(DierctoryType.temp),
    TYPE_FILE(DierctoryType.file),
    TYPE_AUDIO(DierctoryType.audio),
    TYPE_IMAGE(DierctoryType.image),
    TYPE_VIDEO(DierctoryType.video),
    TYPE_THUMB_IMAGE(DierctoryType.thumb),
    TYPE_THUMB_VIDEO(DierctoryType.thumb)
}


@JvmInline
value class DierctoryType(val path:String) {
    companion object{
        val audio = DierctoryType("audio/")
        val data = DierctoryType("data/")
        val file = DierctoryType("file/")
        val log = DierctoryType("log/")
        val cache = DierctoryType("cache/")
        val temp = DierctoryType("temp/")
        val image = DierctoryType("image/")
        val thumb = DierctoryType("thumb/")
        val video = DierctoryType("video/")
    }
}